src/pages/Login/SignUp.js

```diff
@@ -0,0 +1,79 @@
+import { PropTypes } from "prop-types";
+import { useState } from "react";
+import api from "../../api/api";
+import { FetchState } from "../../hooks";
+
+const SignUp = ({ setRegister, dispatch }) => {
+  const [name, setName] = useState();
+  const [email, setEmail] = useState();
+  const [password, setPassword] = useState();
+
+  const handleSignup = async (e) => {
+    e.preventDefault();
+    dispatch({ type: FetchState.FETCH_INIT });
+    try {
+      const user = await api.createAccount(email, password, name);
+      await api.createSession(email, password);
+      dispatch({ type: FetchState.FETCH_SUCCESS, payload: user });
+    } catch (e) {
+      dispatch({ type: FetchState.FETCH_FAILURE });
+    }
+  };
+
+  return (
+    <>
+      <section className="container h-screen mx-auto flex">
+        <div className="flex-grow flex flex-col max-w-xl justify-center p-6">
+          <h1 className="text-6xl font-bold">Sign Up</h1>
+          <p className="mt-4">
+            {" "}
+            Already have an account ?{" "}
+            <a
+              className="cursor-pointer underline"
+              onClick={() => setRegister(false)}
+            >
+              Login
+            </a>{" "}
+          </p>
+          <form onSubmit={handleSignup}>
+            <label className="block mt-6"> Name</label>
+            <input
+              className="w-full p-4 placeholder-gray-400 text-gray-700 bg-white text-lg border-0 border-b-2 border-gray-400 focus:ring-0 focus:border-gray-900"
+              type="text"
+              onChange={(e) => setName(e.target.value)}
+            />
+
+            <label className="block mt-6"> Email</label>
+            <input
+              className="w-full p-4 placeholder-gray-400 text-gray-700 bg-white text-lg border-0 border-b-2 border-gray-400 focus:ring-0 focus:border-gray-900"
+              type="text"
+              onChange={(e) => setEmail(e.target.value)}
+            />
+            <label className="block mt-6"> Password</label>
+            <input
+              className="w-full p-4 placeholder-gray-400 text-gray-700 bg-white text-lg border-0 border-b-2 border-gray-400 focus:ring-0 focus:border-gray-900"
+              type="password"
+              onChange={(e) => setPassword(e.target.value)}
+            />
+
+            <div className="mt-6">
+              <button
+                type="submit"
+                disabled={!name || !email || !password}
+                className="mx-auto mt-4 py-4 px-16 font-semibold rounded-lg shadow-md bg-gray-900 text-white border hover:border-gray-900 hover:text-gray-900 hover:bg-white focus:outline-none disabled:opacity-50 disabled:cursor-not-allowed"
+              >
+                Sign Up
+              </button>
+            </div>
+          </form>
+        </div>
+      </section>
+    </>
+  );
+};
+
+// Login.propTypes = {
```

Comment:
If the code is commented out- remove it. No need to for un-used code

Reply:
Yeap/ I'll do. Thank you so so much for the extensive review. Really appreciate it :D

Reply:
Done